spec/models/fetch_page_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'rails_helper'
+
+RSpec.describe FetchPage, type: :model do
+  describe ".run" do
+    context "when meta tag has charset as utf-8" do
+      it "replace the charset to utf-8" do
+        result = FetchPage.instance.run("http://edition.cnn.com/2016/08/02/politics/obama-says-trump-unfit-for-presidency/index.html")
+        expect(result).to include('utf-8')
+      end
+    end
+
+    context "when meta tag has charset different from utf-8" do
+      it "does not set charset" do
+        result = FetchPage.instance.run("http://www.cdc.gov/ecoli/2009/")
+        expect(result).to include('utf-8')
+      end
+    end
+  end
+
+  describe ".identify_encoding_type" do
+    context "with utf-8 charset" do
```

Comment:
should apply gem 'vcr' for fetching the content of HTTP request http://www.cdc.gov/ecoli/2009/